topology/nhlt/intel/dmic/dmic-process.c

```diff
@@ -0,0 +1,1365 @@
+
+		/* if cfg->mfir_a */
+		if (di == 0) {
+			/* FIR A */
+			fir_decim = MAX(cfg->mfir_a - 1, 0);
+			fir_length = MAX(cfg->fir_a_length - 1, 0);
+			val = FIR_CONTROL_A_START(1) |
+				FIR_CONTROL_A_ARRAY_START_EN(array_a) |
+				FIR_CONTROL_A_DCCOMP(dccomp) |
+				FIR_CONTROL_A_MUTE(fir_mute) |
+				FIR_CONTROL_A_STEREO(stereo[i]);
+			dmic->dmic_blob_fir[i][di].fir_control = val;
+
+			val = FIR_CONFIG_A_FIR_DECIMATION(fir_decim) |
+				FIR_CONFIG_A_FIR_SHIFT(cfg->fir_a_shift) |
+				FIR_CONFIG_A_FIR_LENGTH(fir_length);
+			dmic->dmic_blob_fir[i][di].fir_config = val;
+
+			val = DC_OFFSET_LEFT_A_DC_OFFS(DCCOMP_TC0);
+			dmic->dmic_blob_fir[i][di].dc_offset_left = val;
+
+			val = DC_OFFSET_RIGHT_A_DC_OFFS(DCCOMP_TC0);
+			dmic->dmic_blob_fir[i][di].dc_offset_right = val;
+
+			val = OUT_GAIN_LEFT_A_GAIN(0);
+			dmic->dmic_blob_fir[i][di].out_gain_left = val;
+
+			val = OUT_GAIN_RIGHT_A_GAIN(0);
+			dmic->dmic_blob_fir[i][di].out_gain_right = val;
+
+			/* Write coef RAM A with scaled coefficient in reverse order */
+			length = cfg->fir_a_length;
+			for (j = 0; j < length; j++) {
+				ci = (int32_t)Q_MULTSR_32X32((int64_t)cfg->fir_a->coef[j],
+							     cfg->fir_a_scale, 31,
+							     DMIC_FIR_SCALE_Q, DMIC_HW_FIR_COEF_Q);
+				cu = FIR_COEF_A(ci);
+				/* blob_pdm[i].fir_coeffs[di][j] = cu; */
+				dmic->dmic_fir_array.fir_coeffs[i][di][j] = cu;
+			}
+			dmic->dmic_fir_array.fir_len[0] = length;
+			dmic->dmic_fir_array.fir_len[1] = 0;
+		}
+
+		if (di == 1) {
+			/* FIR B */
+			fir_decim = MAX(cfg->mfir_b - 1, 0);
+			fir_length = MAX(cfg->fir_b_length - 1, 0);
+			val = FIR_CONTROL_B_START(1) |
+				FIR_CONTROL_B_ARRAY_START_EN(array_b) |
+				FIR_CONTROL_B_DCCOMP(dccomp) |
+				FIR_CONTROL_B_MUTE(fir_mute) |
+				FIR_CONTROL_B_STEREO(stereo[i]);
+			dmic->dmic_blob_fir[i][di].fir_control = val;
+
+			val = FIR_CONFIG_B_FIR_DECIMATION(fir_decim) |
+				FIR_CONFIG_B_FIR_SHIFT(cfg->fir_b_shift) |
+				FIR_CONFIG_B_FIR_LENGTH(fir_length);
+			dmic->dmic_blob_fir[i][di].fir_config = val;
+			val = DC_OFFSET_LEFT_B_DC_OFFS(DCCOMP_TC0);
+			dmic->dmic_blob_fir[i][di].dc_offset_left = val;
+
+			val = DC_OFFSET_RIGHT_B_DC_OFFS(DCCOMP_TC0);
+			dmic->dmic_blob_fir[i][di].dc_offset_right = val;
+
+			val = OUT_GAIN_LEFT_B_GAIN(0);
+			dmic->dmic_blob_fir[i][di].out_gain_left = val;
+
+			val = OUT_GAIN_RIGHT_B_GAIN(0);
+			dmic->dmic_blob_fir[i][di].out_gain_right = val;
+
+			/* Write coef RAM B with scaled coefficient in reverse order */
+			length = cfg->fir_b_length;
+			for (j = 0; j < length; j++) {
+				ci = (int32_t)Q_MULTSR_32X32((int64_t)cfg->fir_b->coef[j],
+							     cfg->fir_b_scale, 31,
+							     DMIC_FIR_SCALE_Q, DMIC_HW_FIR_COEF_Q);
+				cu = FIR_COEF_B(ci);
+				/* blob_pdm[i].fir_coeffs[di][j] = cu; */
+				dmic->dmic_fir_array.fir_coeffs[i][di][j] = cu;
+			}
+			dmic->dmic_fir_array.fir_len[1] = length;
+		}
+	}
+
+	return 0;
+}
+
+static struct intel_dmic_params * get_dmic(struct tplg_pre_processor *tplg_pp)
```

Comment:
I think your get and set are inverted. get should be used for reading input values form conf and set should be used to set the values in the generated blob.